pkg/scheduler/framework/framework_test.go

```diff
@@ -410,20 +356,23 @@ func TestUpdateBindingsWithErrors(t *testing.T) {
 		WithScheme(scheme.Scheme).
 		Build()
 
-	var genericUpdateFn = func(ctx context.Context, hubClient client.Client, binding *placementv1beta1.ClusterResourceBinding) error {
-		binding.SetLabels(map[string]string{"test-key": "test-value"})
-		return hubClient.Update(ctx, binding, &client.UpdateOptions{})
+	var genericUpdateFn = func(ctx context.Context, hubClient client.Client, binding placementv1beta1.BindingObj) error {
+		if crb, ok := binding.(*placementv1beta1.ClusterResourceBinding); ok {
```

Comment:
Hi Ryan! Same thing here; we need to verify for both CRBs and RBs I assume?

Reply:
Of course I understand that currently all the UTs care only about CRBs; I guess we could expand the test cases later?

Reply:
yeah, this PR only does refactor and it's already very big. I will leave the following RP PR to add test here if needed. In most cases, I think we don't need to add any resource object UT as the logics are exactly the same.